src/dms-preview/azext_dms/custom.py

```diff
@@ -69,90 +82,85 @@ def create_or_update_project(
         enable_data_integrity_validation=False,
-        enable_query_analysis_validation=False):
-
-    # Set inputs to lowercase
+        enable_query_analysis_validation=False,
+        validate_only=False):
+
+    # Get source and target platform abd set inputs to lowercase
+    src, tgt = get_project_platforms(cmd,
+                                     project_name=project_name,
+                                     service_name=service_name,
+                                     resource_group_name=resource_group_name)
     task_type = task_type.lower()
-    source_platform = source_platform.lower()
-    target_platform = target_platform.lower()
+    scenario_handled_in_extension = extension_handles_scenario(src, tgt, task_type)
 
     # Validation: Test scenario eligibility
-    if not determine_scenario_eligibility(source_platform, target_platform, task_type):
+    if not scenario_handled_in_extension:
         # If not an extension scenario, run CLI core method
-        # TODO: We currently don't have any CLI core code to perform any validations
-        # because of this we need to raise the error here.
         try:
-            # CLI core doesnt currently support task types - it only supports offline migrations.
-            # TODO: Remove this check after task types are supported
-            if source_platform != "sql" or target_platform != "sqldb" or task_type != "offlinemigration":
+            # TODO: Remove this check after validations are added to core
+            if scenario_handled_in_extension:
                 raise ValueError
 
-            core_res = core_create_task(
-                client,
-                resource_group_name,
-                service_name,
-                project_name,
-                task_name,
-                source_connection_json,
-                target_connection_json,
-                database_options_json,
-                enable_schema_validation,
-                enable_data_integrity_validation,
-                enable_query_analysis_validation)
+            core_res = core_create_task(client,
+                                        resource_group_name,
+                                        service_name,
+                                        project_name,
+                                        task_name,
+                                        source_connection_json,
+                                        target_connection_json,
+                                        database_options_json,
+                                        enable_schema_validation,
+                                        enable_data_integrity_validation,
+                                        enable_query_analysis_validation)
         except:
+            # TODO: We currently don't have any CLI core code to perform any validations
+            # because of this we need to raise the error here.
             raise ValueError("The provided source-platform, target-platform, and task-type \
```

Comment:
Since we removed the source/target platforms, I think we can remove this line. (Since the user didn't provide them)

Reply:
Do the same for projects as well.

Reply:
Well the user does provide source and target platforms for a project so I'm going to leave that alone. As for here, I think it would be appropriate to change the wording to something like "The source and target platforms of the parent project are not valid..." something.